File: examples/scala/src/main/scala/com/xebia/functional/xef/scala/auto/Recipe.scala
package com.xebia.functional.xef.scala.auto

import com.xebia.functional.xef.scala.auto.*
import io.circe.Decoder
import com.xebia.functional.xef.prompt.Prompt

private final case class Recipe(name: String, ingredients: List[String]) derives SerialDescriptor, Decoder

@main def runRecipe: Unit =
  conversation {
    val recipe: Recipe = prompt(Prompt("Recipe for chocolate chip cookies."))
    println(s"The recipe for ${recipe.name} is ${recipe.ingredients}")
  }

File: examples/kotlin/src/main/kotlin/com/xebia/functional/xef/auto/Recipe.kt
package com.xebia.functional.xef.auto

import com.xebia.functional.xef.auto.llm.openai.OpenAI
import com.xebia.functional.xef.auto.llm.openai.prompt
import kotlinx.serialization.Serializable

@Serializable data class Recipe(val name: String, val ingredients: List<String>)

suspend fun main() =
  OpenAI.conversation {
    val recipe: Recipe = prompt("Recipe for chocolate chip cookies.")
    println("The recipe for ${recipe.name} is ${recipe.ingredients}")
  }
